data-prepper-api/src/main/java/com/amazon/dataprepper/model/buffer/MultiBufferDecorator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.model.buffer;
+
+import com.amazon.dataprepper.model.CheckpointState;
+import com.amazon.dataprepper.model.record.Record;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Buffer decorator created for pipelines that make use of multiple buffers, such as PeerForwarder-enabled pipelines. The decorator
+ * acts as a pass-through to the primary buffer for all methods except isEmpty, which verifies that all buffers are empty.
+ *
+ * @since 2.0
+ */
+public class MultiBufferDecorator<T extends Record<?>> implements Buffer<T> {
```

Comment:
I think this should be in core. If it is in API, we have a contract for it. I'd rather not have to keep this as something we commit to keeping, but is instead used to help the core use-case.